File: app/src/main/java/sk/styk/martin/apkanalyzer/util/FirstStartHelper.java
package sk.styk.martin.apkanalyzer.util;

import android.content.Context;

/**
 * @author Martin Styk
 * @version 15.11.2017.
 */
public class FirstStartHelper {

    public static final String FIRST_APP_START = "first_app_start";


    public static boolean isFirstStart(Context context) {

        SharedPreferencesHelper preferencesHelper = new SharedPreferencesHelper(context);
        return preferencesHelper.readBoolean(FIRST_APP_START, true);
    }

    public static void setFirstStartFinished(Context context) {
        SharedPreferencesHelper preferencesHelper = new SharedPreferencesHelper(context);
        preferencesHelper.putBoolean(FIRST_APP_START, false);
    }
}

File: app/src/main/java/sk/styk/martin/apkanalyzer/util/FirstStartHelper.kt
package sk.styk.martin.apkanalyzer.util

import android.content.Context
import android.content.Intent
import sk.styk.martin.apkanalyzer.activity.intro.IntroActivity

/**
 * @author Martin Styk
 * @version 15.11.2017.
 */
object FirstStartHelper {

    private const val FIRST_APP_START = "first_app_start"

    /**
     * Execute all actions related to app's first start
     */
    fun execute(context: Context) {
        if (isFirstStart(context))
            context.startActivity(Intent(context, IntroActivity::class.java))
    }

    fun isFirstStart(context: Context): Boolean {
        return SharedPreferencesHelper(context).readBoolean(FIRST_APP_START, true)
    }

    fun setFirstStartFinished(context: Context) {
        SharedPreferencesHelper(context).putBoolean(FIRST_APP_START, false)
    }
}
